UnitsNet.Tests/UnitAbbreviationsCacheTests.cs

```diff
@@ -308,6 +308,15 @@ public void MapUnitToAbbreviation_AddCustomUnit_DoesNotOverrideDefaultAbbreviati
             Assert.Equal("m²", cache.GetDefaultAbbreviation(AreaUnit.SquareMeter));
         }
 
+        [Fact]
+        public void MapUnitToDefaultAbbreviation_AddCustomAbbreviation_UsesNewAbbreviationWithToString()
```

Comment:
`UsesNewAbbreviationWithToString` naming doesn't seem to match what you are testing here. I think you want two test cases here:
1. Rename this test case to `MapUnitToDefaultAbbreviation_GivenUnitAndCulture_SetsDefaultAbbreviationForUnitAndCulture`, and make sure to specify the same culture in `GetDefaultAbbreviation()` call
2. Add new test case `MapUnitToDefaultAbbreviation_GivenCustomAbbreviation_SetsAbbreviationUsedByQuantityToString` that tests how `Area.FromSquareMeters(1).ToString(IFormatProvider)` returns the new custom abbreviation

Reply:
Implementation looks good to me, so just some improvements to tests and this should be good to go 👍

Reply:
Interesting... that new test fails.  Somehow I think you knew it would...

Reply:
Ah... the `ToString()` uses the default cache, so creating a new cache in the test is useless.

Reply:
I didn't :-) The new code structure in 4.0 is still a bit new to me, but yes, these ToString() methods that rely on static types is a bit of a pain to test.